### ISSUE TITLE
Figure out how to handle `batch.remaining: 0` at start of processing

### ISSUE THREAD

**Issue by nathanielrindlaub** [tnc-ca-geo/animl-api#113]
There's a stage early on in the bulk upload lifecycle in which the SQS queue has been built but the images are still getting unzipped and created so there haven't been any messages added to the queue yet. During that stage, when we calculate [`approximateNumberOfMessages`](https://github.com/tnc-ca-geo/animl-api/blob/abeee77289a8b4aa45d92b9ef7b3b0d617a7ea54/src/api/db/models/Batch.js#L62C18-L62C18), SQS correctly returns 0 and we return that to the frontend as `batch.remaining: 0`, but the frontend interprets that meaning that there are no more messages to process, so it currently tells users "Finished <batch.total> of <batch.total> images". 

@ingalls is there a way to keep returning `batch.remaining = null` until `approximateNumberOfMessages` returns a positive value? I suppose we could set some flag in the front end do to this too, but it feels a little clunky.

**Comment by nathanielrindlaub** [comment]
Actually this seems straightforward. We find the `batch` record before querying SQS for `approximateNumberOfMessages`, so if SQS returns 0, before setting remaining to 0 we could just check if the existing `batch.remaining === null` and keep it `null` if so.

**Comment by nathanielrindlaub** [comment]
Nevermind... that won't work b/c `batch.remaining` never gets written to the DB, so there's never anything to compare it against. Needs more thinking...

**Comment by nathanielrindlaub** [comment]
I attempted to solve this by adding two more fields to batch records (`uploadComplete` and `ingestionComplete`) so the front-end has a little more info on where the uploads stand. In chronological order, we:

1. [set `uploadComplete`](https://github.com/tnc-ca-geo/animl-ingest/blob/44fe17b733a6c1dfb30c20d8fbd498c668790647/ingest-zip/index.js#L96) - which indicates that the zip file has been received - right after `ingest-zip` receives the zip file and preprocesses it (this is where we were setting `processingStart` previously).
2. [set `processingStart`](https://github.com/tnc-ca-geo/animl-ingest/blob/44fe17b733a6c1dfb30c20d8fbd498c668790647/ingest-zip/index.js#L116) - after the cloudformation stack has been created. Essentially processing means "we've received the file and provisioned the resources to process it and we've now begun to ingest and process the images"
3. [set `ingestionComplete`](https://github.com/tnc-ca-geo/animl-ingest/blob/44fe17b733a6c1dfb30c20d8fbd498c668790647/ingest-zip/index.js#L166) - which indicates that all images have been saved. 

Most of the time the actual ML processing of the images begins before we set `ingestionComplete`, so I thought this would help solve the problem because we can just show the user a "Saving images..." message until batch returns `ingestionComplete: <date>`, at which point hopefully the app has had enough of a head-start that it's added at least some messages to SQS so `remaining` should no longer be 0. HOWEVER this is not always the case, so this remains an open (albeit smaller than before) issue.

@ingalls when you're back from vacation let me know if you have any ideas for this one.
